Fetches GitHub-Username for easier info-retrieval

heiglandreas opened the pull request:
This PR adds the github-nickname to the users dataset. 

That makes information retrieval via the GitHub-API much easier as currently only the Github-Internal ID is stored which isn't easy to use. If you want to get information about a specific user you can now simply call https://api.github.com/users/[githubName] to get all informations about the user.

Before you had to first call https://api.github.com/users/?per_page=1&since=[githubUid - 1] to get
the nickname and then call again the above users-endpoint to get the infos you might be interested in. 

Therefore you now only need one request instead of two.

This fixes Issue #53

heiglandreas committed:
Fetches GitHub-Username for easier info-retrieval

This commit adds the github-nickname to the users dataset. That makes
information retrieval via the github-API much easier as currently only
the Github-Internal ID is stored which isn't easily used with the
github-API. If you want to get information about a specific user you can
now simply call https://api.github.com/users/<githubName> to get all
inforations about the user.

Before you had to first call
https://api.github.com/users/?per_page=1&since=<githubUid - 1> to get
the username and then call again the above users-endpoint.

christickner commented:
Thanks for this, it will be useful to have the github username.

This PR will get the nickname for *new* users, but not for existing users who login. Because of this the DB will be rather inconsistent because a lot of people have already registered.

What is required to modify AuthController to fetch and save the nickname of a user during login if we don't already have it stored?

heiglandreas commented:
Is that something for the AuthController? Or wouldn't it be easier to once run a (dirty) script that fetches the github-username for all already existing users and inserts it into the database? So from that point on *all* users have a github username set instead of waiting that a user logs in again.

heiglandreas committed:
Adds a command to update missing github-Usernames

This commit adds a console-command that can be used to update all
missing githubUsernames. Call ```app/console
mentoring:addMissingGithubUsernames``` and it will check for empty
githubUsernames and update them accordingly.

dragonmantank commented:
Github usernames can change, so it wouldn't be a bad idea for the auth controller to update that setting at login. It doesn't happen often, but Github does allow you to do it.

At the very least the DB up script would need to include some sort of functionality to update the usernames so everyone has the info.

christickner commented:
+1 on attempting to update username on login and for the upgrade script. If we can get both of them, this is a good step. 

If someone changes their Github username, all that would be necessary is for them to login here and we'd have it updated. That way, we won't need to manually change DB rows and it makes exposing this field in a form unnecessary.

heiglandreas committed:
Adds nickname-retrieval on every login

This commit adds retrieval of the users github-nickname on every login.
So every time a user logs in the github username is fetched and written
to the database.

heiglandreas commented:
That last commit changes (or adds if it's not already present) the users github-nickname on every login.

I'm not sure the script to retrieve ALL github usernames should be part of the DB-up-script as it might take a long time to run. As soon as the UP-script is run the DB is in a state to support the new field in the API but as it is an extension to the API in my eyes there is no need to have the data already available right after the new code has been added. The worst that can happen is that there is no value in the ```githubUsername```-field. The script to retrieve all github-nicknames can then be run and suddenly the values are present. But that's just my 2 cent.

heiglandreas committed:
Fixes a CodeSniffer-Issue

Removed a comment and extended anotherone to bee more clear on what to
do

christickner commented:
I tend to agree that a one-time script that we can run manually is probably sufficient. @dragonmantank is there a reason that it should be a part of the DB migration?

dragonmantank commented:
Only reason I suggested the DB migration is because that script is automatically run during the build process, so the `update()` method could contain the code that gets fired off without us doing anything extra.

If it's a separate script that's not a big deal, we just have to remember to run it by hand on the server.

dragonmantank commented:
Finally got around to merging this in. Thanks!
